lua/lvim/core/terminal.lua

```diff
@@ -77,49 +78,69 @@ local function get_dynamic_terminal_size(direction, size)
   end
 end
 
-M.setup = function()
-  local terminal = require "toggleterm"
-  terminal.setup(lvim.builtin.terminal)
-
-  for i, exec in pairs(lvim.builtin.terminal.execs) do
-    local direction = exec[4] or lvim.builtin.terminal.direction
-
-    local opts = {
-      cmd = exec[1] or lvim.builtin.terminal.shell,
-      keymap = exec[2],
-      label = exec[3],
-      -- NOTE: unable to consistently bind id/count <= 9, see #2146
-      count = i + 100,
-      direction = direction,
-      size = function()
-        return get_dynamic_terminal_size(direction, exec[5])
-      end,
-    }
-
-    M.add_exec(opts)
-  end
-
-  if lvim.builtin.terminal.on_config_done then
-    lvim.builtin.terminal.on_config_done(terminal)
-  end
+local function keybind_toggle(keybind)
+  local Terminal = require("toggleterm.terminal").Terminal
+  local term = Terminal:new(keybind)
+  term:toggle(keybind.size, keybind.direction)
 end
 
-M.add_exec = function(opts)
-  local binary = opts.cmd:match "(%S+)"
+local function add_keybind_keymap(keybind)
```

Comment:
any specific reason for renaming functions? keep in mind that this would break existing API

Reply:
These functions are only used locally, so it makes sense to make them local.  The only code that uses it was also refactored. We could make this one public, but same name or not the opts are now passed directly to toggleterm:new, not the same API
More on the naming, see below